packages/orn-locator/src/resolvers/books.ts

```diff
@@ -0,0 +1,155 @@
+import fetch from 'node-fetch';
+
+const oswebUrl = 'https://openstax.org/apps/cms/api/v2/pages';
+const fields = 'cnx_id,authors,publish_date,cover_color,amazon_link,book_state,promote_image,webview_rex_link,cover_url,title_image_url';
+
+const getArchiveInfo = async (bookId: string) => {
+  const releaseJson = await fetch('https://openstax.org/rex/release.json')
+    .then(response => response.json())
+  ;
+
+  const bookConfig = releaseJson.books[bookId];
+
+  return {
+    archivePath: bookConfig.archiveOverride || releaseJson.archiveUrl,
+    bookVersion: bookConfig.defaultVersion,
+  };
+};
+
+export const book = async(id: string) => {
+  const oswebData = await fetch(`${oswebUrl}?type=books.Book&fields=${fields}&cnx_id=${id}`)
+    .then(response => response.json() as any)
+    .then(data => data.items[0])
+  ;
+
+  return {
+    id,
+    orn: `https://openstax.org/orn/content:book/${id}`,
+    type: 'content:book' as const,
+    title: oswebData.title,
+    slug: oswebData.meta.slug,
+    theme: oswebData.cover_color,
+    urls: {
+      default: oswebData.meta.html_url,
+      information: oswebData.meta.html_url,
+      experience: oswebData.webview_rex_link,
+    },
+    images: {
+      default: oswebData.cover_url,
+      square: oswebData.cover_url,
+      wide: oswebData.title_image_url,
+      promotional: oswebData.promote_image.meta.download_url,
+    }
+  };
+};
+
+const mapTree = (bookId: string) => (tree: any) => {
+  if (tree.contents) {
+    const subTreeId = tree.id.split('@')[0];
+    return {
+      id: subTreeId,
+      title: tree.title,
+      contents: tree.contents.map(mapTree(bookId)),
+      orn: `https://openstax.org/orn/content:collection/${bookId}:${subTreeId}`,
+      type: 'content:collection' as const,
+    };
+  } else {
+    const pageId = tree.id.split('@')[0];
+    return {
+      id: pageId,
+      title: tree.title,
+      orn: `https://openstax.org/orn/content:page/${bookId}:${pageId}`,
+      type: 'content:page' as const,
+    };
+  }
+};
+
+const findTreeNode = (id: string, tree: any): any => {
+  if (tree.id.split('@')[0] === id) {
+    return tree;
+  }
+
+  for (const node of (tree.contents || [])) {
+    const result = findTreeNode(id, node);
+    if (result) {
+      return result;
+    }
+  }
+};
+
+export const bookDetail = async(id: string) => {
```

Comment:
I would call this bookWithContents to make it clear that here is where you get the contents, but if you just want the name to be similar to Rex, that's fine too